components/application-gateway/internal/authorization/csrf/csrf_strategy_test.go

```diff
@@ -0,0 +1,105 @@
+package csrf_test
```

Comment:
Code contained is csrf.go file is tested in csrf_test.go and csrf_strategy_test.go. I think contents of csrf_strategy_test.go could be moved to csrf_test.go.

Reply:
csrf_test.go and csrf_strategy_test.go are in different packages. The csrf_strategy_test.go file was extracted to csrf_test package to avoid circular dependencies. Additionally, we can test the logic from outside its own package.

Reply:
You could solve the circular dependency by moving csrfclient to a separate package. OAuthStrategy is a good example : it is in authorization package and uses client defined in oauth package.

Reply:
@akgalwas moving client won't help here.
Circular dependency happens between the mock, the test file from _csrf_ package and public _*csrf.Response_ type.
It goes as follows: Mock requires *csrf.Response type since it's the part of a method signature being mocked. This type is defined in _csrf_ package. If the test itself is in the same package (_csrf_), there's an inevitable circular dependency. The best way to solve is to put test in a _different_ package, like we did here: _csrf_test_.
This does not happen in oauth package, since the Mock there does not depend on any type from "parent" package.